sql/api/src/main/scala/org/apache/spark/sql/types/StringType.scala

```diff
@@ -31,7 +32,8 @@ class StringType private(val collationId: Int) extends AtomicType with Serializa
   /**
    * Returns whether assigned collation is the default spark collation (UTF8_BINARY).
    */
-  def isDefaultCollation: Boolean = collationId == CollationFactory.DEFAULT_COLLATION_ID
+  def isDefaultCollation: Boolean =
```

Comment:
My fear is that this may be perf sensitive place. can you keep this as simple int compare? `collationNameToId` is dictionary search against string field...